### PR TITLE
Querystring

### PR DESCRIPTION
**Pull request by StanMathers**
Until next release, I'm also going to add new methods to querystring which I found useful when writing examples.

### CODE REVIEW

**Comment by FeodorFitsner** [comment]
Cool! Ready to merge?

**Comment by StanMathers** [comment]
> Cool! Ready to merge?

Oh, wait a minute, I forgot changes with page in controls

**Comment by FeodorFitsner** [comment]
Sure! Let me know when it's ready.

**Comment by StanMathers** [comment]
Done, I added `query` in page properties and linked details to navigation and routing, where I added Querystring with its description, examples and gifs. It's ready to merge.

File: docs/controls/page.md [hunk 1]
```diff
@@ -490,6 +490,14 @@ Note `view=flet.FLET_APP_HIDDEN` which hides app window on start.
 
 A simple PubSub implementation for passing messages between app sessions.
 
+### `query`
+`query` property supports the following methods:
+* go(key: str)
```
Comment (FeodorFitsner) [inline docs/controls/page.md:495]: typo?

File: docs/guides/python/navigation-and-routing.md [hunk 2]
```diff
@@ -190,4 +190,94 @@ To change URL strategy use `route_url_strategy` paremeter of `flet.app()` method
...
+```
+
+### New methods for querystring parsing
+
+* page.query.get(key: str)
+* page.query.to_dict
+* page.query.path
+
+#### Examples
+Lets say we have a URL structure like this
```
Comment (FeodorFitsner) [inline docs/guides/python/navigation-and-routing.md:220]: Leave a space between heading and the next paragraph.

File: docs/guides/python/navigation-and-routing.md [hunk 3]
```diff
@@ -190,4 +190,94 @@ To change URL strategy use `route_url_strategy` paremeter of `flet.app()` method
...
+### Changes with old methods
+
+`go` method of `Page` object now supports `**kwargs` arguments. You can pass key-value pairs to `go` method after the url to receive result like this: `...some_path/?name=John&surname=Doe`
+
+
+Example
+```python
+page.go("/products/", id=5, name_startswith='g')
+```
+Result:
```
Comment (FeodorFitsner) [inline docs/guides/python/navigation-and-routing.md:208]: Add space between lines too.

**Comment by FeodorFitsner** [comment]
Also, `yarn.lock` changed and `package-lock.json` added?

File: docs/controls/page.md [hunk 4]
```diff
@@ -490,6 +490,14 @@ Note `view=flet.FLET_APP_HIDDEN` which hides app window on start.
 
 A simple PubSub implementation for passing messages between app sessions.
 
+### `query`
+`query` property supports the following methods:
```
Comment (FeodorFitsner) [inline docs/controls/page.md:494]: + spacing.

File: docs/guides/python/navigation-and-routing.md [hunk 5]
```diff
@@ -190,4 +190,94 @@ To change URL strategy use `route_url_strategy` paremeter of `flet.app()` method
...
+page.query.path
+```
+Result: '/products/'
+
+### Sample Project
+```python
+import flet
+from flet import *
+import sqlite3
+class Db:
```
Comment (FeodorFitsner) [inline docs/guides/python/navigation-and-routing.md:247]: Insert line between imports and a class definition.

File: docs/guides/python/navigation-and-routing.md [hunk 6]
```diff
@@ -190,4 +190,94 @@ To change URL strategy use `route_url_strategy` paremeter of `flet.app()` method
...
+                       )
+                       """)
+    def get_user_by_id(self, user_id: int) -> str:
+        self.c.execute(f"SELECT name FROM users WHERE id = {user_id}")
+        return [self.c.fetchone()[0]]
+    
+    def get_user_by_start_with(self, start: str) -> str:
+        self.c.execute(f"SELECT name FROM users WHERE name LIKE '{start}%'")
+        return [i[0] for i in self.c.fetchall()]
+def main(page: Page):
```
Comment (FeodorFitsner) [inline docs/guides/python/navigation-and-routing.md:264]: add an empty line before main.

**Comment by StanMathers** [comment]
> Also, `yarn.lock` changed and `package-lock.json` added?

Well, when installing `Docusaurus` locally, it changed that. If there is something important, I might be able to revert `yarn.lock`.

**Comment by FeodorFitsner** [comment]
@StanMathers Would you like to finish this PR?

**Comment by StanMathers** [comment]
> @StanMathers Would you like to finish this PR?

Yup, I'll resume working on it. It'll be done in a few days.

**Pull request closed without merging**